lib/scope.js

```diff
@@ -471,27 +472,34 @@ module.exports = {
       // name is correct every single time we see a signature field. Because
       // of how we must detect the need to hit 'continue', all remaining vars
       // will be looped through and get 'found', removing them from the table.
-      let is_signature_page = await scope.page.$('.dasignature');
-      if ( set_to.includes('/sign') && is_signature_page ) {  // allow sign, signature, signs, etc.
-        a_var_was_found = true;  // Even though it doesn't have a proper var name
-        await scope.sign( scope );
-        // 1 bold, 2 underline, 52 encircled, 53 overlined, 36 blue
-        process.stdout.write(`\x1b[36m${ ' ~ ' }\x1b[0m`);  // signed
-        continue;
-      }
+      if ( set_to.includes('/sign') ) {
+        let is_signature_page = await scope.page.$('.dasignature');
+        if ( is_signature_page ) {
+          a_var_was_found = true;  // Even though it doesn't have a proper var name
+          await scope.sign( scope );
+          process.stdout.write(`\x1b[36m${ '~' }\x1b[0m`);  // signed
+        } else {
+          // Remember any variable that wasn't found yet
+          remaining_vars.push( row );
```

Comment:
Just for my own understanding, does `remaining_vars` act as a queue? That is, it pops an element from the front of the queue and tries to finish the question related to that element. If it can't, it adds the element to the back of the queue.

Reply:
No, it just keeps track of what variables haven't been used yet. Everything stays in the same order. Here's the idea with all the stuff in between removed:
```py
original_list_of_vars = [ 'var1', 'var2' ]
fresh_list_of_vars = []
for a_var in original_list_of_vars:
  if found_on_page( a_var ):
    set_var()
  else:
    fresh_list_of_vars.push( a_var )
```

Does that make sense?